### ISSUE TITLE
idris intepreter’s :doc lacks superclass feature

### ISSUE THREAD

**Issue by phaazon** [idris-lang/Idris-dev#2659]
`:doc Applicative` doesn’t show the `Functor` superclass, for instance. We need that!

**Comment by david-christiansen** [comment]
Just for the reference of whoever finds this and may want to work on it, `:doc` does show subclasses. It should be fairly straightforward to also look the other direction and include that output.

**Comment by phaazon** [comment]
I’m trying to write a patch for that. It’s my first time with Idris, so I don’t really know how to do that but I’ve found the code where it’s defined. I’ll keep you informed!

**Comment by phaazon** [comment]
Reading at the code, I guess the output is correctly set, but it sounds that the superclasses in `ClassInfo` is always empty. That’s weird, I can’t find out where that information is put (in the `Ctxt` type).

**Comment by david-christiansen** [comment]
I just took a peek at it, and it seems that `ClassInfo` doesn't represent the necessary information. Sorry to have mislead you! I think the solution is to add a `superclasses` field to `ClassInfo` that the class elaborator sticks in there, then use it for docs, but it seems that's a bit more difficult that we had hoped.

**Comment by phaazon** [comment]
I have a few spare time, I’ll try to figure it out. Thanks!

**Comment by phaazon** [comment]
Yeah, it’s actually harder than I expected. I should learn from scratch because I’m dangling around and don’t really know where to find the information – especially where the `IState` is populated.

**Comment by libeako** [comment]
Similar behavior : the generated html documentation of an interface does not mention the super interfaces.
